core/drivers/imx_lpuart.c

```diff
@@ -63,10 +64,72 @@ KEEP_PAGER(imx_lpuart_ops);
 void imx_uart_init(struct imx_uart_data *pd, paddr_t base)
 {
 	pd->base.pa = base;
+	pd->base.va = 0;
 	pd->chip.ops = &imx_lpuart_ops;
 
 	/*
 	 * Do nothing, debug uart(sc lpuart) shared with normal world,
 	 * everything for uart initialization is done in bootloader.
 	 */
 }
+
+#ifdef CFG_DT
+
+static struct serial_chip *imx_lpuart_dev_alloc(void)
+{
+	struct imx_uart_data *pd = malloc(sizeof(struct imx_uart_data));
```

Comment:
`sizeof(*pd)` please, and `calloc()` to zero-initialize.

Reply:
Same comment applies to `imx_uart.c`, too.

Reply:
I will fix it.

Reply:
I will fix it.